src/panels/lovelace/common/handle-action.ts

```diff
@@ -50,17 +50,28 @@ export const handleAction = async (
   ) {
     forwardHaptic("warning");
 
+    let serviceName;
+    if (actionConfig.action === "call-service") {
+      const [domain, service] = actionConfig.service.split(".", 2);
+      const serviceDomains = hass.services;
+      if (domain in serviceDomains && service in serviceDomains[domain]) {
+        serviceName = serviceDomains[domain][service].name;
+      }
+    }
+
     if (
       !(await showConfirmationDialog(node, {
         text:
           actionConfig.confirmation.text ||
           hass.localize(
             "ui.panel.lovelace.cards.actions.action_confirmation",
             "action",
-            hass.localize(
-              "ui.panel.lovelace.editor.action-editor.actions." +
-                actionConfig.action
-            ) || actionConfig.action
+            serviceName ||
+              hass.localize(
+                "ui.panel.lovelace.editor.action-editor.actions." +
+                  actionConfig.action
+              ) ||
```

Comment:
We still want to show this part I think, just add what service will be called?

Reply:
Also, I think many names assume that they will be used together with the domain/integration name, and not stand alone?

Reply:
I think the end-user does not care if it is technically a "service" or not that is being called. So I opted only for the name.

I agree reg. the domain part. Need to see how to best include that.

Reply:
@bramkragten I attempted to take over the logic that is also used in the ha-service-picker for the domain. The problem is that I cannot get the translations to get loaded. So the first time the popup gets shown, the domain is not translated, the second time around it is.

I attempted to manually force the "title" loading via `hass.loadBackendTranslation("title");` but I did not have any effect.

=> Any hints / tips?

Reply:
Probably a timing issue then? Do you await it? And use its result as input for `domainToName`?

Reply:
I am sure I did, but apparently both individually (either await or local var) and not combined (await + local var). Tried that now, and seems to work:

![image](https://user-images.githubusercontent.com/114137/112981491-9141eb80-915b-11eb-9e7e-ca0d7b32f7b9.png)